contracts/mainnet/connectors/notional/main.sol

```diff
@@ -655,6 +655,139 @@ abstract contract NotionalResolver is Events, Helpers {
+	{
+		if (noteAmount == type(uint256).max)
+			noteAmount = note.balanceOf(address(this));
+
+		if (ethAmount == type(uint256).max) ethAmount = address(this).balance;
+
+		approve(note, address(staking), noteAmount);
+
+		staking.mintFromETH{ value: ethAmount }(noteAmount, minBPT);
+
+		_eventName = "LogMintSNoteFromETH(address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(address(this), ethAmount, noteAmount, minBPT);
+	}
+
+	/// @notice Mints sNOTE from some amount of NOTE and WETH
+	/// @dev Mints sNOTE from some amount of NOTE and WETH
+	/// @param noteAmount amount of NOTE to transfer into the sNOTE contract
+	/// @param wethAmount amount of WETH to transfer into the sNOTE contract
+	/// @param minBPT slippage parameter to prevent front running
+	function mintSNoteFromWETH(
+		uint256 noteAmount,
+		uint256 wethAmount,
+		uint256 minBPT
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		if (noteAmount == type(uint256).max)
+			noteAmount = note.balanceOf(address(this));
+
+		if (wethAmount == type(uint256).max)
+			wethAmount = weth.balanceOf(address(this));
+
+		approve(note, address(staking), noteAmount);
+		approve(weth, address(staking), wethAmount);
+
+		staking.mintFromWETH(noteAmount, wethAmount, minBPT);
+
+		_eventName = "LogMintSNoteFromWETH(address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(address(this), noteAmount, wethAmount, minBPT);
+	}
+
+	/// @notice Begins a cool down period for the sender, this is required to redeem tokens
+	/// @dev Current cool down period is set to 1296000 seconds (15 days)
+	function startCoolDown()
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		staking.startCoolDown();
+
+		_eventName = "LogStartCoolDown(address)";
+		_eventParam = abi.encode(address(this));
+	}
+
+	/// @notice Stops a cool down for the sender
+	/// @dev User must start another cool down period in order to call redeemSNote
+	function stopCoolDown()
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		staking.stopCoolDown();
+
+		_eventName = "LogStopCoolDown(address)";
+		_eventParam = abi.encode(address(this));
+	}
+
+	/// @notice Redeems some amount of sNOTE to underlying constituent tokens (ETH and NOTE).
+	/// @dev An account must have passed its cool down expiration before they can redeem
+	/// @param sNOTEAmount amount of sNOTE to redeem
+	/// @param minWETH slippage protection for ETH/WETH amount
+	/// @param minNOTE slippage protection for NOTE amount
+	/// @param redeemWETH true if redeeming to WETH to ETH
+	function redeemSNote(
+		uint256 sNOTEAmount,
+		uint256 minWETH,
+		uint256 minNOTE,
+		bool redeemWETH
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		if (sNOTEAmount == type(uint256).max)
+			sNOTEAmount = staking.balanceOf(address(this));
+
+		approve(staking, address(staking), sNOTEAmount);
```

Comment:
Is this approval required? I don't think so right?

Reply:
remvoed